datafusion/core/src/physical_optimizer/aggregate_statistics.rs

```diff
@@ -148,10 +152,10 @@ fn take_optimizable_table_count(
                 .as_any()
                 .downcast_ref::<expressions::Literal>()
             {
-                if lit_expr.value() == &ScalarValue::UInt8(Some(1)) {
+                if lit_expr.value() == &COUNT_STAR_EXPANSION {
```

Comment:
There was an implicit coupling between the SQL planner and this file, which I have now made explicit with a named constant